### PR TITLE
Add AmountDetails <AmtDtls> to EntryDetails

### PR DESCRIPTION
**Pull request by antlih**
For getting AmtDtls->TxAmt->Amt. 

Tested with CAMT 053.001.02 and 054.001.02. 

Feel free to modify and give feedback.

### CODE REVIEW

File: src/Decoder/EntryTransactionDetail.php [hunk 1]
```diff
@@ -206,6 +209,37 @@ public function addBankTransactionCode(DTO\EntryTransactionDetail $detail, Simpl
     }
 
     /**
+     * @param DTO\EntryTransactionDetail $detail
+     * @param SimpleXMLElement           $xmlDetail
+     * @param SimpleXMLElement           $CdtDbtInd
+     */
+    public function addAmountDetails(DTO\EntryTransactionDetail $detail, SimpleXMLElement $xmlDetail, $CdtDbtInd)
+    {
+
```
Comment (Einenlum) [inline src/Decoder/EntryTransactionDetail.php]: Useless blank line here :)

File: src/Decoder/EntryTransactionDetail.php [hunk 2]
```diff
@@ -206,6 +209,37 @@ public function addBankTransactionCode(DTO\EntryTransactionDetail $detail, Simpl
...
+        if (isset($xmlDetail->AmtDtls)) {
+            $amountDetails = new DTO\AmountDetails();
+
+            if (isset($xmlDetail->AmtDtls->TxAmt) && isset($xmlDetail->AmtDtls->TxAmt->Amt)) {
+
```
Comment (Einenlum) [inline src/Decoder/EntryTransactionDetail.php]: Useless blank line here :)

File: src/Decoder/EntryTransactionDetail.php [hunk 3]
```diff
@@ -206,6 +209,37 @@ public function addBankTransactionCode(DTO\EntryTransactionDetail $detail, Simpl
...
+                $amount = StringToUnits::convert((string) $xmlDetail->AmtDtls->TxAmt->Amt);
+                
+                if ((string) $CdtDbtInd === 'DBIT') {
+                    $amount = $amount * -1;
+                }
+
+                $money = new Money(
+                                    $amount,
```
Comment (Einenlum) [inline src/Decoder/EntryTransactionDetail.php]: bad indent

**Comment by Einenlum** [comment]
Thanks! I just added a few CS comments.

File: src/Decoder/EntryTransactionDetail.php (hunk 3 above)
Comment (antlih) [inline src/Decoder/EntryTransactionDetail.php]: Ok, how would you ident this?

Comment (Einenlum) [inline src/Decoder/EntryTransactionDetail.php]: Well, not that far ^^

```php
public function addAmountDetails(DTO\EntryTransactionDetail $detail, SimpleXMLElement $xmlDetail, $CdtDbtInd)
{
    if (isset($xmlDetail->AmtDtls)) {
        $amountDetails = new DTO\AmountDetails();

        if (isset($xmlDetail->AmtDtls->TxAmt) && isset($xmlDetail->AmtDtls->TxAmt->Amt)) {
            $amount = StringToUnits::convert((string) $xmlDetail->AmtDtls->TxAmt->Amt);
            
            if ((string) $CdtDbtInd === 'DBIT') {
                $amount = $amount * -1;
            }

            $money = new Money(
                $amount,
                new Currency((string) $xmlDetail->AmtDtls->TxAmt->Amt['Ccy'])
            );
            $amountDetails->setAmount($money);
        }
        $detail->setAmountDetails($amountDetails);
    }
}
```

**Pull request merged**